### DISCUSSION TITLE
BackgroundTasks Integration Tests

### DISCUSSION THREAD

**Discussion by artiom-gesp** [fastapi/fastapi#9088] [Questions] [answered]
I want to be able to create an **integration** test based on an endpoint that creates a background task.
The task function receives a session object and will be used to create/update rows in a database.

What is the best practice for it?
Is it possible to not depend on time.sleep ? I also don't want to directly use a db session in case we change our SQLAlchemy ORM to an async ORM in the future (I don't want this to affect the tests).

Thank you in advance! By the way, awesome community!

### Example

```Python
from fastapi import FastAPI, Depends

app = FastAPI()

def write(session, name):
    // Add a row to the database.

@app.get("/")
def read_root(task: BackgroundTasks, session = Depends(get_session):
    task.add_task(write, session, "name") // will be executed after return
    return {"success": True}
```

### Environment

* FastAPI Version 0.58.1:

**Comment by rspadim** [comment, marked as answer]
Mock the add task, and call the api using testclient from fastapi
